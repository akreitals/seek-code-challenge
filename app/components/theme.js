const maxContainerWidth = '940px';

const emoji = '"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"';
const font = `Averta, -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Helvetica, Arial, sans-serif, ${emoji}`;
const mono = 'SFMono-Regular,"Roboto Mono",Menlo,monospace';

const fontSizes = [12, 14, 16, 18, 24, 32, 48, 64, 72, 96];
const space = [0, 4, 8, 16, 32, 64, 128];
const regular = 400;
const bold = 600;

const greys = {
    black: '#1c1c1c',
    charcoal: '#404040',
    midGrayDark: '#747474',
    midGrayMedium: '#898989',
    midGray: '#a8a8a8',
    midGrayLight: '#d6d6d6',
    grayMedium: '#e1e1e1',
    grayLight: '#eeeeee',
    grayLightest: '#f7f7f7',
    offWhite: '#fbfbfb',
    white: '#ffffff'
};

const brand = {
    blueDarker: '#071d40',
    blueDark: '#042763',
    blue: '#0d3880',
    blueLight: '#184da8',
    blueLighter: '#2765cf',
    pink: '#e60278',
    green: '#157e00',
    greenLight: '#5bc252',
    purple: '#9556b7',
    teal: '#00a5ad',
    orange: '#f57c00',
    yellow: '#ffc600',
    yellowLight: '#fffce3',

    primary: '#e60278',
    accent: '#9556b7',
    success: '#157e00',
    info: '#2765cf',
    warning: '#f57c00',
    error: '#e60278'
};

export const colors = {
    ...greys,
    ...brand
};

export const radius = '2px';

// boxShadows
export const boxShadows = [
    `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
];

const theme = {
    colors,
    font,
    mono,
    fontSizes,
    space,
    regular,
    bold,
    maxContainerWidth,
    radius,
    boxShadows
};

export default theme;
