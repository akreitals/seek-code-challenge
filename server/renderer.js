import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
