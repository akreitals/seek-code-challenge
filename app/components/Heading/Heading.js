import Text from '../Text/Text';
import theme from '../theme';

const Heading = Text.withComponent('h3').extend`
    line-height: 1.25;
`;

Heading.displayName = 'Heading';

Heading.defaultProps = {
    regular: true,
    fontSize: 4,
    m: 0,
    theme
};

Heading.h1 = Heading.withComponent('h1');
Heading.h1.defaultProps = {
    fontSize: 6,
    regular: true,
    m: 0
};

Heading.h2 = Heading.withComponent('h2');
Heading.h2.defaultProps = {
    fontSize: 5,
    bold: true,
    m: 0
};

Heading.h3 = Heading.withComponent('h3');
Heading.h3.defaultProps = {
    fontSize: 4,
    bold: true,
    m: 0
};

Heading.h4 = Heading.withComponent('h4');
Heading.h4.defaultProps = {
    fontSize: 3,
    regular: true,
    m: 0
};

Heading.h5 = Heading.withComponent('h5');
Heading.h5.defaultProps = {
    fontSize: 2,
    bold: true,
    m: 0
};

Heading.h6 = Heading.withComponent('h6');
Heading.h6.defaultProps = {
    fontSize: 0,
    bold: true,
    uppercase: true,
    m: 0
};

export default Heading;
