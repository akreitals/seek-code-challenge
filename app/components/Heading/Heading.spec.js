import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme, { colors } from '../theme';
import Heading from './Heading';

describe('Heading', () => {
    test('should render a <Heading /> component', () => {
        const json = renderer.create(<Heading />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('should render with h1-h6 dot-notation', () => {
        const headingH1json = renderer.create(<Heading.h1 />).toJSON();
        const headingH2json = renderer.create(<Heading.h2 />).toJSON();
        const headingH3json = renderer.create(<Heading.h3 />).toJSON();
        const headingH4json = renderer.create(<Heading.h4 />).toJSON();
        const headingH5json = renderer.create(<Heading.h5 />).toJSON();
        const headingH6json = renderer.create(<Heading.h6 />).toJSON();
        expect(headingH1json).toMatchSnapshot();
        expect(headingH2json).toMatchSnapshot();
        expect(headingH3json).toMatchSnapshot();
        expect(headingH4json).toMatchSnapshot();
        expect(headingH5json).toMatchSnapshot();
        expect(headingH6json).toMatchSnapshot();
    });

    test('should render with default theme font size, when fontSize prop not used', () => {
        const json = renderer.create(<Heading />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('font-size', `${theme.fontSizes[4]}px`);
    });

    test('should render with <Text> component properties', () => {
        const json = renderer
            .create(<Heading textAlign="center" color="primary" />)
            .toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('text-align', 'center');
        expect(json).toHaveStyleRule('color', colors.primary);
    });
});
