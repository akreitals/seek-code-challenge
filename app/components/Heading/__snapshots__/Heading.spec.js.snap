// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a <Heading /> component 1`] = `
.c0 {
  font-size: 24px;
  margin: 0px;
  font-weight: 400;
  line-height: 1.25;
}

<h3
  className="c0"
  fontSize={4}
/>
`;

exports[`Heading should render with <Text> component properties 1`] = `
.c0 {
  font-size: 24px;
  margin: 0px;
  color: #e60278;
  text-align: center;
  font-weight: 400;
  line-height: 1.25;
}

<h3
  className="c0"
  color="primary"
  fontSize={4}
/>
`;

exports[`Heading should render with default theme font size, when fontSize prop not used 1`] = `
.c0 {
  font-size: 24px;
  margin: 0px;
  font-weight: 400;
  line-height: 1.25;
}

<h3
  className="c0"
  fontSize={4}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 1`] = `
.c0 {
  font-size: 48px;
  margin: 0px;
  font-weight: 400;
  line-height: 1.25;
}

<h1
  className="c0"
  fontSize={6}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 2`] = `
.c0 {
  font-size: 32px;
  margin: 0px;
  font-weight: 600;
  line-height: 1.25;
}

<h2
  className="c0"
  fontSize={5}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 3`] = `
.c0 {
  font-size: 24px;
  margin: 0px;
  font-weight: 600;
  line-height: 1.25;
}

<h3
  className="c0"
  fontSize={4}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 4`] = `
.c0 {
  font-size: 20px;
  margin: 0px;
  font-weight: 400;
  line-height: 1.25;
}

<h4
  className="c0"
  fontSize={3}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 5`] = `
.c0 {
  font-size: 16px;
  margin: 0px;
  font-weight: 600;
  line-height: 1.25;
}

<h5
  className="c0"
  fontSize={2}
/>
`;

exports[`Heading should render with h1-h6 dot-notation 6`] = `
.c0 {
  font-size: 12px;
  margin: 0px;
  font-weight: 600;
  text-transform: uppercase;
  line-height: 1.25;
}

<h6
  className="c0"
  fontSize={0}
/>
`;
